core/templates/dev/head/exploration_editor/settings_tab/SettingsTab.js

```diff
@@ -40,6 +40,9 @@ oppia.controller('SettingsTab', [
       EXPLORATION_TITLE_INPUT_FOCUS_LABEL);
 
     $scope.CATEGORY_LIST_FOR_SELECT2 = [];
+    //Empty category added to force dropdown to scroll top
```

Comment:
@seanlip @souravbadami if we go with option 2 (always unshift), the fix will be only changing "push" for "unshift"

In this example I added "Hello". the new category appears at the top.

![image](https://cloud.githubusercontent.com/assets/2034660/15759916/c05c65ac-28c5-11e6-9db9-128b892ac75b.png)